include/quicr/fetch_track_handler.h

```diff
@@ -0,0 +1,164 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024 Cisco Systems
+// SPDX-License-Identifier: BSD-2-Clause
+
+#pragma once
+
+#include <quicr/detail/base_track_handler.h>
+#include <quicr/detail/messages.h>
+#include <quicr/metrics.h>
+
+namespace quicr {
+    /**
+     * @brief
+     */
+    class FetchTrackHandler : public BaseTrackHandler
+    {
+      public:
+        /**
+         * @brief Receive status codes
+         */
+        enum class Error : uint8_t
+        {
+            kOk = 0,
+            kNotAuthorized,
+            kObjectNotAvailable,
+        };
+
+        /**
+         * @brief  Status codes for the track
+         */
+        enum class Status : uint8_t
+        {
+            kOk = 0,
+            kNotConnected,
+            kFetchError,
+            kNotAuthorized,
+            kObjectNotAvailable,
```

Comment:
Considering start/end object isn't part of send fetch, is this used for group not available?

Reply:
How is it handled when some groups are available but the complete range is not? for example, fetch group 10 - 20, 10 - 15 are available, but 16 - 20 are not. Does the status change after receiving all of group 15?

I think what we want in this case is not a partial fetch.  I think if the fetch cannot be completely satisfied for the ranges specified, it should maybe return `'kGroupRangeNotAvailable` and `kObjectRangeNotAvailable`. In terms of MoQ, not really a good architecture because a suggested range that could be satisfied is not returned in the error response.

Reply:
`¯\_(ツ)_/¯`